libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightAlertConfig/AlertAttributeSelecxt.tsx

```diff
@@ -20,279 +20,263 @@ import {
 } from "../../../../constants/index.js";
 import { IExecutionResultEnvelope } from "../../../../../model/index.js";
 
-import { getAttributeTitle } from "./utils.js";
-import {
-    AttributeValue,
-    useAttributeValuesFromExecResults,
-} from "./hooks/useAttributeValuesFromExecResults.js";
+import { useAttributeValuesFromExecResults } from "./hooks/useAttributeValuesFromExecResults.js";
+import { getSelectedCatalogAttribute, getSelectedCatalogAttributeValue } from "./utils/getters.js";
 
 export interface IAlertAttributeSelectProps {
     execResult: IExecutionResultEnvelope | undefined;
     selectedAttribute: AlertAttribute | undefined;
     selectedValue: string | null | undefined;
     onAttributeChange: (attribute: AlertAttribute | undefined, value: string | undefined) => void;
     attributes: AlertAttribute[];
-    catalogAttributes: IAttributeMetadataObject[];
+    catalogAttributes: ICatalogAttribute[];
+    catalogDateDatasets: ICatalogDateDataset[];
 }
 
 export const AlertAttributeSelect = ({
     execResult,
-    selectedAttribute,
+    selectedAttribute: sa,
     selectedValue,
     onAttributeChange,
     attributes,
     catalogAttributes,
+    catalogDateDatasets,
 }: IAlertAttributeSelectProps) => {
     const intl = useIntl();
     const ref = useRef<HTMLElement | null>(null);
 
+    const availableAttributes = useMemo(() => {
+        return attributes.filter((attr) => attr.type === "attribute");
+    }, [attributes]);
+
     const [searchString, setSearchString] = useState("");
     const [isOpen, setIsOpen] = useState(false);
     const [isOpenAttribute, setIsOpenAttribute] = useState<string | null>(null);
 
     const { isResultLoading, getAttributeValues } = useAttributeValuesFromExecResults(execResult);
 
-    const selectedAttr = useMemo(
-        () => selectedAttribute && getSelectedCatalogAttribute(catalogAttributes, selectedAttribute),
-        [catalogAttributes, selectedAttribute],
-    );
+    const selectedAttribute = useMemo(() => {
+        return sa && getSelectedCatalogAttribute(catalogAttributes, catalogDateDatasets, sa);
+    }, [sa, catalogAttributes, catalogDateDatasets]);
     const selectedVal = useMemo(
```

Comment:
Here again, to make it more straightforward, would be using some convention for the name, rather than `selectedVal`